src/flisp/julia_extensions.c

```diff
@@ -152,22 +152,55 @@ value_t fl_julia_identifier_start_char(fl_context_t *fl_ctx, value_t *args, uint
     return jl_id_start_char(wc) ? fl_ctx->T : fl_ctx->F;
 }
 
-// return NFC-normalized UTF8-encoded version of s
+#include "julia_charmap.h"
+#define _equal_wchar_(x, y, ctx) ((x) == (y))
+#define _hash_wchar_(x, ctx) inthash((uint32_t) ((uintptr_t) (x)))
+#include "htable.inc"
+HTIMPL_R(wcharhash, _hash_wchar_, _equal_wchar_)
+
+void jl_charmap_init(fl_context_t *fl_ctx)
+{
+    size_t charmap_len = sizeof(charmap) / (2*sizeof(uint32_t));
+    size_t i;
+    htable_t *h = htable_new(&fl_ctx->jl_charmap, charmap_len);
+    assert(sizeof(uint32_t) <= sizeof(void*));
+    for (i = 0; i < charmap_len; ++i) {
+        /* Store charmap in a hash table.  Typecasting codepoints
+           directly to pointer keys works because pointers are at
+           least 32 bits on all Julia-supported systems, and because
+           we never map anything to U+0001 (since HT_NOTFOUND is (void*)1). */
+        assert((void*)charmap[i][1] != HT_NOTFOUND);
+        wcharhash_put_r(h, (void*)((uintptr_t)charmap[i][0]),
+                           (void*)((uintptr_t)charmap[i][1]), (void*)fl_ctx);
+    }
+}
+utf8proc_int32_t jl_charmap_map(utf8proc_int32_t c, void *fl_ctx_)
+{
+    fl_context_t *fl_ctx = (fl_context_t *) fl_ctx_;
+    htable_t *h = &fl_ctx->jl_charmap;
+    void *v = wcharhash_get_r(h, (void*)((uintptr_t)c), (void*) fl_ctx);
+    return v == HT_NOTFOUND ? c : (utf8proc_int32_t) ((uintptr_t) v);
+}
+
+// return NFC-normalized UTF8-encoded version of s, with
+// additional custom normalizations defined by jl_charmap above.
 static char *normalize(fl_context_t *fl_ctx, char *s)
 {
     // options equivalent to utf8proc_NFC:
     const int options = UTF8PROC_NULLTERM|UTF8PROC_STABLE|UTF8PROC_COMPOSE;
     ssize_t result;
     size_t newlen;
-    result = utf8proc_decompose((uint8_t*) s, 0, NULL, 0, (utf8proc_option_t)options);
+    result = utf8proc_decompose_custom((uint8_t*) s, 0, NULL, 0, (utf8proc_option_t)options,
```

Comment:
It could be confusing to have Julia builds with normalization and others without. Personally, for RPM packages, I'll also provide a new version of utf8proc where needed (even if it would of course be simpler not to have to do this).